Documentation/components/drivers/character/watchdog.rst

```diff
@@ -153,4 +153,51 @@ This command registers an user callback that will be triggered on timeout. It re
 
 .. c:macro:: WDIOC_KEEPALIVE
 
- This command resets the watchdog timer ("ping",  "pet the dog",  "feed the dog"). 
\ No newline at end of file
+ This command resets the watchdog timer AKA '**ping**", "**kick**", "**pet** the dog",  "**feed** the dog"). 
+
+Using the Auto-monitor feature to reset the watch dog,
+------------------------------------------------------
+
+The auto-monitor provides an OS-internal mechanism to automatically start and repeatedly reset the watchdog.  
+
+To enable it, follow the next instructions:
+
+1. Select a Watchdog Timer Instance
+
+ To select the wdt browse in the ``menuconfig`` using the following path:
+
+ Go into menu :menuselection:`System Type --> <Chip> Peripheral Selection` and press :kbd:`Enter`. Then select one watchdog timer.
+
+2. Enable the Auto-monitor option
+
+ Go into menu :menuselection:`Device Drivers --> Timer Driver Support` and press :kbd:`Enter`. Then enable:
+
+ - [x] Watchdog Timer Support
+
+ Then press :kbd:`Enter` again to enter into the Watchdog Timer Support menu. And finally enable the Auto-monitor option:
+
+ - [x] Auto-monitor
+
+ After selecting the option you may want to configure some parameters:
+
+ * **Timeout**: It is the watchdog timer expiration time in seconds.
+ * **Keep a live interval**: This is the interval in which the watchdog will be fed. It is in seconds. It can't be bigger than the timeout. If this interval is equal to timeout interval, than this interval will automatically change to half timeout. 
+ * **Keep alive by**: This is a choice to determine who is going to feed the dog. There are 4 possible choices that are described as follows.
+
+ ``Capture callback``: This choice registers a watchdog timer callback to reset the watchdog every time it expires, i.e., on timeout. 
+
+ ``Timer callback``: This choice also uses a timer callback to reset the watchdog, but it will reset the watchdog every "keep a live interval".
+
+ ``Worker callback``: This choice uses the Low Priority Work Queue to reset the watchog every "keep a live interval". This choice depends on having the Low Priority Work Queue enabled. 
```

Comment:
not really need Low Priority Work Queue:

1. If only high work enable, the high work queue is used
2. otherwise the low work queue is used

This is because, high work queue will map to low work queue if only high work queue is enabled.

Reply:
Hum, so I'll change and add it as follows:

``Worker callback``: This choice uses a Work Queue to reset the watchdog every "keep a live interval". This choice depends on having the Low or High Priority Work Queue enabled.
If only the High Priority Work Queue is enabled, this one will be used, otherwise Low Priority Work Queue is used.

What do you think?

BTW, I thought it was only LPWORK because of the configuration:
```
      work_queue(LPWORK, &upper->work, watchdog_automonitor_worker,
                 upper, WATCHDOG_AUTOMONITOR_PING_INTERVAL);
```

I didn't know the LPWORK was replaced by the HPWORK in case the latter was enabled and the former was not.

Reply:
Why invent terms? Keep alive -> Watchdog Reset interval

Reply:
@saramonteiro yes, it's accurate now, thanks.

Reply:
@davids5 I did not create terms like "Keep alive" and "pet the dog" or "ping". 
Actually I just used the terms I saw in `drivers/timers/watchdog.c` and in `include/nuttx/timers/watchdog.h`
For example, the ioctl that feeds the dog is actually named `WDIOC_KEEPALIVE`.
My main goal with this PR is to document the WDT related features sticking to the current terms used in the source code.

Reply:
@xiaoxiang781216 thank you! =)

Reply:
> @davids5 I did not create terms like "Keep alive" and "pet the dog" or "ping".
> Actually I just used the terms I saw in `drivers/timers/watchdog.c` and in `include/nuttx/timers/watchdog.h`
> For example, the ioctl that feeds the dog is actually named `WDIOC_KEEPALIVE`.
> My main goal with this PR is to document the WDT related features sticking to the current terms used in the source code.

Fare enough.